zephyr/lib/cpu.c

```diff
@@ -218,10 +218,7 @@ void cpu_disable_core(int id)
 	}
 #if defined(CONFIG_PM)
 	/* TODO: before requesting core shut down check if it's not actively used */
-	if (!pm_state_force(id, &(struct pm_state_info){PM_STATE_SOFT_OFF, 0, 0})) {
-		tr_err(&zephyr_tr, "failed to set PM_STATE_SOFT_OFF on core %d", id);
-		return;
-	}
+	pm_state_force(id, &(struct pm_state_info){PM_STATE_SOFT_OFF, 0, 0});
```

Comment:
yeah, but the function returns a "success status" supposedly. Should we add an assert in case the implementation changes? And in production builds it will be removed anyway

Reply:
I think its wise keeping the check - its safe for future API changes, and its non time critical code.

Reply:
ok ill close this PR. Anyway, I want to remove the call for pm_state_force() during disable and use sys_poweroff() instead.